Document/0x05g-Testing-Network-Communication.md

```diff
@@ -117,6 +117,14 @@ Certificate pinning is the process of associating the backend server with a part
 
 The certificate can be pinned and hardcoded into the app or retrieved at the time the app first connects to the backend. In the latter case, the certificate is associated with ("pinned" to) the host when the host is seen for the first time. This alternative is less secure because attackers intercepting the initial connection can inject their own certificates.
 
+Note that there are various ways to work with a failing pin:
+
+- Inform the user about not being able to connect to the back-end and stop all operations. The app can check whether there is an update and inform the user about updating to the latest version of the app if available. 
+  - Additionally a call can be made to the back-end or to a crash-reporting service that the pinning has failed to inform the developer and/or back-end of the attack or misconfiguration.
+- Only inform the back-end by doing a call with slightly different parameters, which automatically informs the back-end that pinning failed. From here on, only a limited amounts of APIs should be available which do not involve processing information that has a higher risk rating.
```

Comment:
As we are not explaining this mechanism (call with slightly different parameters), what would be the difference to the previous way? 

Maybe we rephrase to make it easier:
- After calling the API or back-end and notify about the failing pinning, the app can still offer limited functionality that shouldn't involve sensitive functions or processing of sensitive data. The communication would happen without SSL Pinning and just validate the X.509 certificate accordingly. 

Does that make sense?

Reply:
LGTM 

" and **notifying** about the failing...

Reply:
There would be a differnece in TLS handshake params, let me check how to get that done properly. The idea would be that we remain using TLS, but with some different data on the handshake, combining it with your feedback now 👍